### PR TITLE
docs: fix outdated commands in release tooling

### PR DESCRIPTION
**Pull request by jhchabran**
I went through the release tooling today (see https://sourcegraph.sourcegraph.com/notebooks/Tm90ZWJvb2s6MTI3) and noticed that a few things in the template were outdated. 

## Test plan

docs

### CODE REVIEW

File: dev/release/templates/release_issue_template.md [hunk 1]
```diff
@@ -164,6 +164,7 @@ On the day of the release, confirm there are no more release-blocking issues (as
 - [ ] Alert the marketing team in [#release-post](https://sourcegraph.slack.com/archives/C022Y5VUSBU) that they can merge the release post.
 - [ ] Finalize and announce that the release is live:
   ```sh
+  pnpm run release release:finalize
```
Comment (jhchabran) [inline dev/release/templates/release_issue_template.md:167]: @coury-clark it's the right place to put that, right?

Comment (coury-clark) [reply dev/release/templates/release_issue_template.md:167]: This is fine, but it ends up being repeated from the src/src PR template
https://github.com/sourcegraph/sourcegraph/pull/55364

**Review by coury-clark — APPROVED** [review]

**Review by mucles — APPROVED** [review]
LGTM , thank you @jhchabran

File: dev/release/templates/release_issue_template.md (hunk 1 above)
Comment (jhchabran) [reply dev/release/templates/release_issue_template.md:167]: Ah! I missed it being over there as well. Thanks 🙏

**Pull request merged**